frame/staking/src/mock.rs

```diff
@@ -233,6 +237,12 @@ parameter_types! {
 	pub static MaxUnlockingChunks: u32 = 32;
 	pub static RewardOnUnbalanceWasCalled: bool = false;
 	pub static LedgerSlashPerEra: (BalanceOf<Test>, BTreeMap<EraIndex, BalanceOf<Test>>) = (Zero::zero(), BTreeMap::new());
+	pub static OnStakeUpdate: Vec<(AccountId, Option<Stake<AccountId, Balance>>)> = Vec::new();
```

Comment:
```
enum Event {
	StakeUpdate(..),
	NominatorUpdate(..),
	...
}
```

and so on might be cleaner all in all.

Reply:
and in either case, please use all of this stuff to a new `parameter_types! {}` above `EventListenerMock`.

Reply:
Yes, seeing the rest of this code, you probably want to do this.

Reply:
Yeah, it’s not very dry. It’s also the approach I initially wanted to use in terms of event emission, like having a single method ‘emit’ in Staking and various events, but I think we agreed on it being too verbose. It certainly works for tests though!